lib/HLSL/HLOperationLower.cpp

```diff
@@ -4421,6 +4421,40 @@ static Value* SkipAddrSpaceCast(Value* Ptr) {
   return Ptr;
 }
 
+// For known non-groupshared, verify that the destination param is valid
+void ValidateAtomicDestination(CallInst *CI, HLObjectOperationLowerHelper *pObjHelper) {
+  Value *dest = CI->getArgOperand(HLOperandIndex::kInterlockedDestOpIndex);
+  GetElementPtrInst *gep = nullptr;
+
+  // If we encounter a gep, we may provide a more specific error message
+  while (isa<GetElementPtrInst>(dest)) {
```

Comment:
If it's at all possible for the base pointer to be global, this should use `GEPOperator` instead, but comment indicates that this function is only called for non-groupshared case, so that ptr should always originate from an intrinsic call I think.
This was just a passing thought, probably nothing to change here.

Reply:
There's no way for correct code to make the base pointer global here. It shouldn't be possible at all, but if it does somehow, this will throw an error at least.